packages/cli-kit/src/private/node/conf-store.ts

```diff
@@ -48,3 +57,30 @@ export function removeSession(config: Conf<ConfSchema> = cliKitStore()): void {
   outputDebug(outputContent`Removing session store...`)
   config.delete('sessionStore')
 }
+
+/**
+ * Fetch from cache, or run the provided function to get the value, and cache it
+ * before returning it.
+ * @param key - The key to use for the cache.
+ * @param fn - The function to run to get the value to cache.
+ * @param timeout - The number of milliseconds to cache the value for.
+ * @returns The value from the cache or the result of the function.
+ */
+export async function cacheFetch(
```

Comment:
Since we are already using classes for the store instances, have we considered having a subclass for the CLIKit Store fixing it to the CLIKit schema? Then we can move this function to that class and the ergonomics would improve:

```js
cliKitStore().cacheFetch()
```

I'd also suggest renaming the function to something that's more explicit about what it does:

```js
// Inspired by ActiveRecord APIs
store.fetchOrInvalidateFromCache()
```

Reply:
Alright, I went with `cacheRetrieveOrRepopulate` which is maybe better.

Also, I don't currently see the major value over the current organization. I think in the past I felt that using classes on conf store ended up making the code more complex, not less.